pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/participantFiles/DownloadParticipantFileRoute.java

```diff
@@ -0,0 +1,73 @@
+package org.broadinstitute.dsm.route.participantFiles;
+
+import static org.broadinstitute.dsm.statics.DBConstants.FILE_DOWNLOAD_ROLE;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.Optional;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.util.SecurityUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import spark.QueryParamsMap;
+import spark.Request;
+
+@Slf4j
+public class DownloadParticipantFileRoute extends RequestHandler {
+    private static String googleProjectName;
+
+    public DownloadParticipantFileRoute(String googleProjectName) {
+        this.googleProjectName = googleProjectName;
+    }
+
+    @Override
+    protected Object processRequest(Request request, spark.Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm = null;
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        }
+        if (StringUtils.isBlank(realm)) {
+            response.status(500);
+            throw new RuntimeException("Realm was missing from the request");
+        }
+        String userIdR = UserUtil.getUserId(request);
```

Comment:
I think we already have `userId` in the arguments of `processRequest`?

Reply:
We check that these two match in `UserUtil.checkUserAccess`